tests/config-array.test.js

```diff
@@ -1291,6 +1291,240 @@ describe('ConfigArray', () => {
 				expect(configs.isFileIgnored(path.join(basePath, 'foo/a.js'))).to.be.false;
 			});
 
+			it('should return false when file has the same name as a directory that is ignored by a pattern that ends with `/`', () => {
+				configs = new ConfigArray([
+					{
+						files: ['**/foo']
+					},
+					{
+						ignores: [
+							'foo/'
+						]
+					}
+				], {
+					basePath
+				});
+
+				configs.normalizeSync();
+
+				expect(configs.isFileIgnored(path.join(basePath, 'foo'))).to.be.false;
+			});
+
+			it('should return false when file is in the parent directory of directories that are ignored by a pattern that ends with `/`', () => {
+				configs = new ConfigArray([
+					{
+						files: ['**/*.js']
+					},
+					{
+						ignores: [
+							'foo/*/'
+						]
+					}
+				], {
+					basePath
+				});
+
+				configs.normalizeSync();
+
+				expect(configs.isFileIgnored(path.join(basePath, 'foo/a.js'))).to.be.false;
+			});
+
+			it('should return true when file is in a directory that is ignored by a pattern that ends with `/`', () => {
+				configs = new ConfigArray([
+					{
+						files: ['**/*.js']
+					},
+					{
+						ignores: [
+							'foo/'
+						]
+					}
+				], {
+					basePath
+				});
+
+				configs.normalizeSync();
+
+				expect(configs.isFileIgnored(path.join(basePath, 'foo/a.js'))).to.be.true;
+			});
+
+			it('should return true when file is in a directory that is ignored by a pattern that does not end with `/`', () => {
+				configs = new ConfigArray([
+					{
+						files: ['**/*.js']
+					},
+					{
+						ignores: [
+							'foo'
+						]
+					}
+				], {
+					basePath
+				});
+
+				configs.normalizeSync();
+
+				expect(configs.isFileIgnored(path.join(basePath, 'foo/a.js'))).to.be.true;
+			});
+
+			it('should return false when file is in a directory that is ignored and then unignord by pattern that end with `/`', () => {
+				configs = new ConfigArray([
+					{
+						files: ['**/*.js']
+					},
+					{
+						ignores: [
+							'foo/',
+							'!foo/'
+						]
+					}
+				], {
+					basePath
+				});
+
+				configs.normalizeSync();
+
+				expect(configs.isFileIgnored(path.join(basePath, 'foo/a.js'))).to.be.false;
+			});
```

Comment:
These three tests were already passing before the change. I added them just to verify that the cases the transformations were targeting still work as intended when the transformations are removed.